.github/workflows/build.yml

```diff
@@ -0,0 +1,93 @@
+name: Github Actions Build
+
+on: [push, pull_request]
+
+jobs:
+  tests:
+    name: ${{ matrix.name }} ${{ matrix.compiler }} ${{ matrix.build_type }} - ${{ matrix.lib_type }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        compiler: [GCC, Clang, MSVC]
+        build_type: [Debug, Release]
+        build_shared: [ON, OFF]
+        exclude:
+          - os: windows-latest
+            compiler: GCC
+          - os: ubuntu-latest
+            compiler: MSVC
+          - os: macos-latest
+            compiler: MSVC
+        include:
+          - compiler: gcc
+            c_compiler: gcc-9
+            cxx_compiler: g++-9
+          - compiler: clang
+            c_compiler: clang
+            cxx_compiler: clang++
+            msvc_toolset: ClangCL
+          - compiler: msvc
+            msvc_toolset: v142
+          - os: macos-latest
+            name: MacOS
+          - os: ubuntu-latest
+            name: Ubuntu
+          - os: windows-latest
+            name: Windows
+          - build_shared: ON
+            lib_type: Shared
+          - build_shared: OFF
+            lib_type: Static
+
+    steps:
+      - uses: lukka/get-cmake@v3.19.2
+      - uses: seanmiddleditch/gha-setup-ninja@v3
+      - uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Buildtools Version
+        run: ${{ matrix.cxx_compiler }} --version
+        if: matrix.os != 'windows-latest'
+
+      - name: Buildtools Version
+        run: |
+          if [ "${{ matrix.compiler }}" == "Clang" ]; then
+            clang-cl.exe --version
+          else
+            cl.exe
+          fi
+        shell: bash
+        if: matrix.os == 'windows-latest'
+
+      - name: Configure
+        run: |
+          mkdir build
+          cd build
+          if [ "$RUNNER_OS" == "Windows" ]; then
+            cmake .. -DINICPP_BUILD_TESTS=ON -DINICPP_BUILD_EXAMPLES=ON -DINICPP_BUILD_SHARED=${{matrix.build_shared}} -T${matrix.msvc_toolset}
```

Comment:
Variable `matrix.msvc_toolset` enclosed in only one pair of braces, that's why Windows builds are failing

Reply:
Oh lol, I just added that after making the PR, previously it was just running with cl.exe on both the configs and you can see there that the runs that build a shared library fail and the runs that build a static library succeed.

Reply:
Fixed in latest commit